woocommerce/woocommerce
# Fix - TGM Plugin activation calling undefined function wc_get_screen_ids

If we add woocommerce and enable the `force_activation` config in TGM plugin activation then the bug appears.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





![image](https://github.com/woocommerce/woocommerce/assets/3774827/6746ed94-eb51-4629-8999-0ec8103cdb7d)



### How to test the changes in this Pull Request:

All this does is ensure that a particular file is loaded before a function within that file is called. Most of the time, the file is already loaded, so in those cases, this change does nothing. Rather than having to install and set up the third-party TGM Plugin, all we really need to do here is make sure this change doesn't break anything. So after switching to this branch, try a few things:

* Deactivating and reactivating the plugin
* Navigate to a few different WooCommerce pages within WP Admin

Doing these things, you shouldn't see any fatal errors about notices or about functions already being defined.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Ensure WC admin functions are defined before attempting to add notices.

#### Comment 

</details>
